src/arrayMethodSort.js

```diff
@@ -21,19 +21,35 @@ function applyCustomSort() {
       throw new TypeError('compareFunction must be a function');
     }
 
-    for (let i = 0; i < this.length; i++) {
-      for (let j = i + 1; j < this.length; j++) {
-        if (compareFunction(this[i], this[j]) > 0) {
-          const temp = this[i];
-
-          this[i] = this[j];
-          this[j] = temp;
-        }
-      }
-    }
+    quickSort(this, compareFunction, 0, this.length - 1);
 
     return this;
   };
 }
 
+function quickSort(array, compareFunction, left, right) {
+  if (left < right) {
```

Comment:
with this you also avoided partinioning the array unnecessarily I think (when it's perfectly ordered)), nice!